### PR TITLE
Clarify samples for push metrics and return likelihood as well

### PR DESCRIPTION
**Pull request by michaelneale**

### CODE REVIEW

File: examples/current_time_data [hunk 1]
```diff
@@ -2,6 +2,5 @@
 
 # another sample script to pump in data from the current time
 # Usage: ./currente_time_data localhost 153
-DOCKER=$1
```
Comment (allanino) [inline examples/current_time_data]: Please keep this argument. I like to use these scripts in Linux, so it's better to be able to specify the host.

File: examples/loop [hunk 2]
```diff
@@ -2,7 +2,10 @@
 
 for i in `seq 1 1000000`;
 do
-	./current_time_data $i
+    
+        ./current_time_data $(uptime | awk '{print $11}')
```
Comment (allanino) [inline examples/loop]: I suggest also setting `DOCKER=$1` in this script, passing it as argument to `./current_time_data`. Also, I needed to change some things in this uptime parser to get it working (like replacing comma for dot in the value). As it's a nice way to get started with what NuPIC is doing, you could change this to something like:
```
#!/bin/bash
DOCKER=$1
for i in `seq 1 1000000`;
do
  ./current_time_data $DOCKER $(uptime | awk '{print $11}' | sed 's/,/\./' | cut -d ',' -f 1)
  echo ""
  sleep 1
done
```

File: monitor/run_monitor_dyn.py [hunk 3]
```diff
@@ -29,16 +31,39 @@
 # the current configured monitors - each one has a chunk of CLA memory
 current_monitors = {}
 
+# track the last time something was seen
+last_seen_input = {}
+
```
Comment (allanino) [inline monitor/run_monitor_dyn.py]: Just a style suggestion: please, remove these double newlines.

File: monitor/run_monitor_dyn.py [hunk 4]
```diff
@@ -29,16 +31,39 @@
...
 def get_monitor(check_id, config):
     """ get or create a new monitor with optional config """
+    last_seen_input[check_id] = time.time()
 
     if check_id not in current_monitors:
         current_monitors[check_id] = new_monitor(check_id, config)
     return current_monitors[check_id]
 
+def garbage_collect():
```
Comment (allanino) [inline monitor/run_monitor_dyn.py]: That's a nice addition. I think it's easier to control things using parameters in here, like:
```
def garbage_collect(timeout):
    """ garbage collect checks that havent' seen action in a while to save memory """
    for check_id in last_seen_input:
        if (time.time() - last_seen_input[check_id] > timeout):
          logger.info("Garbage collecting: %s", check_id)
          remove_monitor(check_id)

def gc_task(cleanup_interval=60, timeout=15):
    """ schedule a regular clean out of garbage """
    def do_gc():
        while True: 
          garbage_collect(timeout)
          time.sleep(cleanup_interval)
    #do some stuff
    worker = threading.Thread(target=do_gc, args=[])
    worker.start()
```

**Comment by michaelneale** [comment]
good suggestions - should get this done today. This is all working quite nicely. I am tempted to add some extra json to the response - with some recent history that a client could do as a little "sparkline" graph as evidence why it thinks there is an anomaly - but I will do that as a separate PR/feature.

File: examples/loop (hunk 2 above)
Comment (michaelneale) [inline examples/loop]: I think I will get rid of that uptime thing - or move to another script. Was just me noodling on my mac. Interesting that uptime value is quite smooth - and oddly the likelihood score was more volatile than anomaly (but not much difference in effect if you are tripping an alert). Ran it for hours to see what happened.

**Comment by michaelneale** [comment]
@allanino ok made those changes - should look better now.

**Comment by allanino** [comment]
Thanks! 

I read somewhere that you would like to keep the uptime loop in a separate script. I agree with that, so that we would have by now two examples: the linear example (were we found those issues) and the system load example using uptime. Maybe we should call them `linear_loop` and `system_load_loop`. In this way it will be even easier for starters to get interesting things running.

**Comment by michaelneale** [comment]
OK fixed. I renamed the loop as well. The uptime script isn't portable between os-x and linux. 

I am looking at another one that perhaps simulates queue depth - which I think is a great demo.

**Comment by allanino** [comment]
Thanks!

**Pull request merged**